mixins/imageHelper.ts

```diff
@@ -64,14 +96,153 @@ declare module 'vue/types/vue' {
         maxWidth,
         maxHeight,
         quality,
-        limitByWidth
+        limitByWidth,
+        backdropIndex,
+        checkParent
       }: ImageUrlForElementParams
-    ) => string;
+    ) => string | undefined;
+    getParentId(item: BaseItemDto): string | undefined;
+    getImageTag(
+      item: BaseItemDto | BaseItemPerson,
+      type: ImageType,
+      index?: number,
+      checkParent?: boolean
+    ): string | undefined;
+    getBlurhash(
+      item: BaseItemDto | BaseItemPerson,
+      type: ImageType,
+      index?: number,
+      checkParent?: boolean
+    ): string | undefined;
   }
 }
 
+/**
+ * Checks if the passed object is BaseItemDto or BaseItemPerson
+ *
+ * @param {BaseItemDto|BaseItemPerson} obj - The object to check
+ * @returns {boolean} Returns true if the object is a person, false otherwise.
+ */
+function isPerson(obj: BaseItemDto | BaseItemPerson): obj is BaseItemPerson {
+  return (obj as BaseItemPerson)?.Role !== undefined;
+}
+
+const excludedBlurhashTypes = [ImageType.Logo];
+
 const imageHelper = Vue.extend({
   methods: {
+    /**
+     * Gets the tag of the image of an specific item and type.
+     *
+     * @param {BaseItemDto|BaseItemPerson} item - The item object.
+     * @param {ImageType} type - The type of the image requested.
+     * @param {number} [index=0] - Index of the Backdrop image (when ImageType equals to 'Backdrop').
+     * @param {boolean} [checkParent=true] - Looks for tag/image type for the parent if the passed item doesn't have the ImageType requested
+     * @returns {string|undefined} Returns the tag, undefined if the specific ImageType doesn't exist.
+     */
+    getImageTag(
+      item: BaseItemDto | BaseItemPerson,
+      type: ImageType,
+      index = 0,
+      checkParent = true
+    ): string | undefined {
+      if (item) {
+        if (!isPerson(item)) {
+          if (item.ImageTags?.[type]) {
```

Comment:
I've rewritten this. Feel free to use if it if you think it is a bit easier to understand:

```
      if (!item) return;

      if (isPerson(item)) {
        if (item.PrimaryImageTag && type === ImageType.Primary) {
          return item.PrimaryImageTag;
        } else {
          return;
        }
      }

      if (item.ImageTags?.[type]) {
        return item.ImageTags?.[type];
      } else if (
        type === ImageType.Backdrop &&
        item.BackdropImageTags?.[index]
      ) {
        return item.BackdropImageTags[index];
      }

      if (checkParent) {
        switch (type) {
          case ImageType.Primary:
            if (item.AlbumPrimaryImageTag) {
              return item.AlbumPrimaryImageTag;
            } else if (item.ChannelPrimaryImageTag) {
              return item.ChannelPrimaryImageTag;
            } else if (item.ParentPrimaryImageTag) {
              return item.ParentPrimaryImageTag;
            }
            break;
          case ImageType.Art:
            if (item.ParentArtImageTag) {
              return item.ParentArtImageTag;
            }
            break;
          case ImageType.Backdrop:
            if (
              item.ParentBackdropImageTags &&
              item.ParentBackdropImageTags[index]
            ) {
              return item.ParentBackdropImageTags[index];
            }
            break;
          case ImageType.Logo:
            if (item.ParentLogoImageTag) {
              return item.ParentLogoImageTag;
            }
            break;
          case ImageType.Thumb:
            if (item.ParentThumbImageTag) {
              return item.ParentThumbImageTag;
            }
        }
      }
```

Reply:
Thank you very much for proposing this, don't know why but I didn't think about using an switch here and it's much clear 👍